src/core/lib/iomgr/iomgr_posix.cc

```diff
@@ -85,7 +85,8 @@ void grpc_set_default_iomgr_platform() {
 }
 
 bool grpc_iomgr_run_in_background() {
-  return grpc_event_engine_run_in_background();
+  return grpc_core::IsEventEngineCallbackCqEnabled() ||
+         grpc_event_engine_run_in_background();
```

Comment:
why isn't the `grpc_event_engine_run_in_background` check enough?

Reply:
See https://github.com/grpc/grpc/blob/1178b2d229ad7fd27df743176dbc2ea104d6b28e/src/core/lib/iomgr/ev_epoll1_linux.cc#L1300

That flag is false for all iomgr "event_engines" (not to be confused with the new EventEngine). This PR bypasses that check, since the iomgr pollers aren't needed for callback execution when using the EventEngine.